File: Year02/2nd semester/MPP/Lab/mpp-proiect-csharp-AlexandruPaul21/CrossPlatform/services/IProjectServices.cs
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace services
{
    public interface IProjectServices
    {
        void login(Agency agency, IProjectObserver client);
        void logout(Agency agency, IProjectObserver client);
        Flight[] getAllFlights(Agency agency, IProjectObserver client);
        Flight[] getFlightsForDestinationAndDate(Flight flight, IProjectObserver client);
        int getNumberOfSeats(Flight flight, IProjectObserver client);
        void bookFlight(Booking booking, IProjectObserver client);
    }
}

File: Year02/2nd semester/MPP/Lab/mpp-proiect-java-AlexandruPaul21/JavaProjectNetworking/ProjectServices/src/main/java/org/example/IProjectServices.java
package org.example;

public interface IProjectServices {
    void login(Agency agency, IProjectObserver client) throws ProjectException;
    void logout(Agency agency, IProjectObserver client) throws ProjectException;
    Flight[] getAllFlights(Agency agency, IProjectObserver client) throws ProjectException;
    Flight[] getFlightsForDestinationAndDate(Flight flight, IProjectObserver client) throws ProjectException;
    int getNumberOfSeats(Flight flight, IProjectObserver client) throws ProjectException;
    void bookFlight(Booking booking, IProjectObserver client) throws ProjectException;
}
